grpc/grpc.go

```diff
@@ -0,0 +1,198 @@
+package grpc
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/jzelinskie/cobrautil/v2"
+
+	"github.com/go-logr/logr"
+	"github.com/jzelinskie/stringz"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/keepalive"
+)
+
+// ConfigureFunc is a function used to configure this CobraUtil
+type ConfigureFunc = func(cu *CobraUtil)
+
+// New creates a configuration that exposes RegisterFlags and RunE
+// to integrate with cobra
+func New(serviceName string, configurations ...ConfigureFunc) *CobraUtil {
+	cu := CobraUtil{
+		serviceName:    stringz.DefaultEmpty(serviceName, "grpc"),
+		preRunLevel:    0,
+		logger:         logr.Discard(),
+		defaultAddr:    ":50051",
+		defaultEnabled: false,
+		flagPrefix:     "grpc",
+	}
+	for _, configure := range configurations {
+		configure(&cu)
+	}
+	return &cu
+}
+
+// CobraUtil carries the configuration for a otel CobraRunFunc
+type CobraUtil struct {
```

Comment:
I'd name this type config in each package instead of cobrautil. I think it makes sense to also make the type private, since all of the fields are also private. I'm pretty sure you can still return the type from functions, but folks will just be unable to construct it from a literal in other packages.

Picking up optgen might also be a good idea for this too: https://github.com/ecordell/optgen